recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,270 @@
+            "meteringmarketplace",
+            "mobileanalytics",
+            "monitoring",
+            "mq",
+            "mturk-requester",
+            "neptune",
+            "opsworks",
+            "opsworkscm",
+            "organizations",
+            "pinpoint",
+            "polly",
+            "pricing",
+            "queues",
+            "quicksight",
+            "ram",
+            "rds",
+            "redshift",
+            "recognition",
+            "resource-groups",
+            "robomaker"
+            "route53",
+            "route53domains",
+            "s3",
+            "sagemaker",
+            "sdb",
+            "serverlessrepo"
+            "servicecatalog",
+            "servicediscovery",
+            "shield",
+            "signer",
+            "sms",
+            "snowball",
+            "sns",
+            "sqs",
+            "ssm",
+            "states",
+            "storagegateway",
+            "sts",
+            "support",
+            "swf",
+            "text-to-speech",
+            "texttract",
+            "transcribe",
+            "transfer",
+            "translate",
+            "waf",
+            "workdocs",
+            "worklink",
+            "workmail",
+            "workspaces",
+            "xray"
+           )
+    options = {
+            **{ x: [True, False] for x in sdks},
+            **{
+                "shared": [True, False],
+                "fPIC": [True, False],
+                "min_size": [True, False],
+                }
+            }
+    default_options = {
+            **{ x: False for x in sdks},
+            **{
+                "shared": False,
+                "fPIC": True,
+                "min_size": False
+                }
+            }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("aws-c-common/0.4.25")
+        self.requires("aws-c-event-stream/0.1.5")
+        self.requires("zlib/1.2.11")
+        if self.settings.os != "Windows":
+            self.requires("libcurl/7.71.1")
+        if not self.settings.os in ["Windows", "Macos"]:
+            self.requires("openssl/1.1.1i")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        # Keeping these unused folders make the source copy fail on Windows
+        shutil.rmtree(self._source_subfolder + "/code-generation/generator")
+        shutil.rmtree(self._source_subfolder + "/aws-cpp-sdk-core-tests")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        build_only = list([])
```

Comment:
having an empty `build_only` by default will make all components to be built which seems counterintuitive with the fact the default options are false.
I suggest replacing this with
```suggestion
        build_only = ['core']
```